File: Projeto Didio Pizza/Didio Pizza Springboot/backend/src/main/java/com/pizzaria/didiopizza/controllers/AdminController.java
package com.pizzaria.didiopizza.controllers;

import java.util.Map;

import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RestController;

import com.pizzaria.didiopizza.service.AdminService;

@RestController
public class AdminController {
    private final AdminService service;

    public AdminController(AdminService service) {
        this.service = service;
    }

    @PostMapping("/admin")
    public ResponseEntity<?> criar(
            @RequestBody Map<String, Object> body) {
        return ResponseEntity.status(201).body(Map.of("id", service.criarAdmin(body)));
    }

    @PostMapping("/admin/login")
    public Map<String, Object> login(
            @RequestBody Map<String, Object> body) {
        return Map.of(
                "token", service.loginAdmin(body),
                "tipo", "admin",
                "nome", "Admin"
        );
    }
}

File: Projeto Didio Pizza/Didio Pizza Node/backend/src/controller/adminController.js
import { Router } from 'express';
import * as adminService from '../service/adminService.js';

const endpoints = Router();


endpoints.post('/admin', async (req, resp) => {
    try {
        let admin = req.body;

        let id = await adminService.criarAdmin(admin);

        resp.status(201).send({
            id: id
        });

    } catch (err) {
        resp.status(400).send({
            erro: err.message
        });
    }
});


endpoints.post('/admin/login', async (req, resp) => {
    try {
        let admin = req.body;

        let token = await adminService.loginAdmin(admin);

        resp.status(200).send({
            token: token
        });

    } catch (err) {
        resp.status(400).send({
            erro: err.message
        });
    }
});


export default endpoints;